pypeit/spectrographs/spectrograph.py

```diff
@@ -1062,16 +1062,20 @@ def select_detectors(self, subset=None):
         if subset is None:
             return np.arange(1, self.ndet+1).tolist()
 
-        if isinstance(subset, str):
-            _subset = parse.parse_slitspatnum(subset)[0].tolist()
+        # Parse subset if it's a string (single slitspatnum) or a list of slitspatnums
+        if isinstance(subset, str) or \
```

Comment:
I don't follow why this is a spectrograph specific functionality. Since it does not appear in any other spectrograph class, it is not spectrograph specific. So shouldn't this go in utils?